src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -177,1327 +205,1182 @@ static std::vector<rct_widget> _window_editor_object_selection_widgets = {
+                    std::reverse(_listItems.begin(), _listItems.end());
+                }
             }
-            auto ft = Formatter();
-            ft.Add<const char*>(gCommonStringFormatBuffer);
-            DrawTextEllipsised(dpi, screenCoords, width_limit, STR_STRING, ft, { colour, fontSpriteBase });
         }
-        screenCoords.y += SCROLLABLE_ROW_HEIGHT;
-    }
-}
-
-static void WindowEditorObjectSetPage(rct_window* w, int32_t page)
-{
-    if (w->selected_tab == page)
-        return;
-
-    w->selected_tab = page;
-    w->selected_list_item = -1;
-    w->scrolls[0].v_top = 0;
-    w->frame_no = 0;
-
-    if (page == EnumValue(ObjectType::Ride))
-    {
-        _listSortType = RIDE_SORT_TYPE;
-        _listSortDescending = false;
-    }
-    else
-    {
-        _listSortType = RIDE_SORT_RIDE;
-        _listSortDescending = false;
+        Invalidate();
     }
 
-    VisibleListRefresh(w);
-    w->Invalidate();
-}
-
-static void WindowEditorObjectSelectionSetPressedTab(rct_window* w)
-{
-    for (size_t i = 0; i < std::size(ObjectSelectionPages); i++)
+    void VisibleListDispose()
     {
-        w->pressed_widgets &= ~(1 << (WIDX_TAB_1 + i));
+        _listItems.clear();
+        _listItems.shrink_to_fit();
     }
-    w->pressed_widgets |= 1LL << (WIDX_TAB_1 + w->selected_tab);
-}
 
-/**
- * Takes the y coordinate of the clicked on scroll list
- * and converts this into an object selection.
- * Returns the position in the list.
- * Object_selection_flags, installed_entry also populated
- *
- *  rct2: 0x006AA703
- */
-static int32_t GetObjectFromObjectSelection(ObjectType object_type, int32_t y)
-{
-    int32_t listItemIndex = y / SCROLLABLE_ROW_HEIGHT;
-    if (listItemIndex < 0 || static_cast<size_t>(listItemIndex) >= _listItems.size())
-        return -1;
 
-    return listItemIndex;
-}
 
-/**
- *
- *  rct2: 0x006D33E2
- */
-static void WindowEditorObjectSelectionManageTracks()
-{
-    set_every_ride_type_invented();
-    set_every_ride_entry_invented();
-
-    gEditorStep = EditorStep::DesignsManager;
-
-    int32_t entry_index = 0;
-    for (; object_entry_get_chunk(ObjectType::Ride, entry_index) == nullptr; entry_index++)
-        ;
-
-    rct_ride_entry* ride_entry = get_ride_entry(entry_index);
-    uint8_t ride_type = ride_entry_get_first_non_null_ride_type(ride_entry);
-
-    auto intent = Intent(WC_TRACK_DESIGN_LIST);
-    intent.putExtra(INTENT_EXTRA_RIDE_TYPE, ride_type);
-    intent.putExtra(INTENT_EXTRA_RIDE_ENTRY_INDEX, entry_index);
-    context_open_intent(&intent);
-}
-
-/**
- *
- *  rct2: 0x006ABBBE
- */
-static void EditorLoadSelectedObjects()
-{
-    auto& objManager = OpenRCT2::GetContext()->GetObjectManager();
-    int32_t numItems = static_cast<int32_t>(object_repository_get_items_count());
-    const ObjectRepositoryItem* items = object_repository_get_items();
-    for (int32_t i = 0; i < numItems; i++)
+    /**
+     *
+     *  rct2: 0x006ABBBE
+     */
+    void EditorLoadSelectedObjects()
```

Comment:
This function doesn't really belong to the window. I suggest it is moved out of the window struct.